go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -34,12 +34,18 @@ type PluginEntry struct {
 	DefaultForTaskTypes []pluginsCore.TaskType
 	// Returns a new KubeClient to be used instead of the internal controller-runtime client.
 	CustomKubeClient func(ctx context.Context) (pluginsCore.KubeClient, error)
+}
+
+// System level properties that this Plugin supports
+type PluginProperties struct {
 	// Disables the inclusion of OwnerReferences in kubernetes resources that this plugin is responsible for.
 	// Disabling is only useful if resources will be created in a remote cluster.
 	DisableInjectOwnerReferences bool
 	// Boolean that indicates if finalizer injection should be disabled for resources that this plugin is
 	// responsible for.
 	DisableInjectFinalizer bool
```

Comment:
Since we are introducing `PluginProperties` for k8s plugins I think moving these properties here make sense